tango_ros_common/tango_ros_common/src/main/java/eu/intermodalics/tango_ros_common/ParameterNode.java

```diff
@@ -64,6 +64,31 @@ public void onStart(ConnectedNode connectedNode) {
         syncLocalPreferencesWithParameterServer();
     }
 
+    // When using simulated time, direct or indirect calls to "connectedNode.getCurrentTime()" such
+    // as connectedNode.getLog().error(e) throw NullPointerExeption.
+    // See rosjava issue reported here: https://github.com/rosjava/rosjava_core/issues/148
+    // As a workaround, this function waits until "connectedNode.getCurrentTime()" stops trowing
+    // this exception, which is the case when:
+    // * simulated time is not used or
+    // * simulated time is used AND /clock is being published
+    // If simulated is used AND /clock is NOT published, it keeps on waiting and logging a warning.
+    public void waitForClock() {
+        while (true) {
+            try {
+                mConnectedNode.getCurrentTime();
+                android.util.Log.i("waitForClock", "Successfully get current time.");
```

Comment:
If simulated time is used (/use_sim_time is set to true) it waits until /clock is published. Otherwise the function returns in ~100ms.